src/reduceProps.js

```diff
@@ -0,0 +1,18 @@
+import curry from 'lodash/function/curry';
+import wrapDisplayName from './wrapDisplayName';
+import createElement from './createElement';
+
+const reduceProps = (reduce, initialState, BaseComponent) => {
+  let state = initialState;
```

Comment:
This value is shared between all instances of ReduceProps. You need it to be local to each component. The only way to do this is with a class component.